WalletWasabi.Daemon/Global.cs

```diff
@@ -452,6 +477,11 @@ public async Task DisposeAsync()
 
 				if (TorManager is { } torManager)
 				{
+					if (OnionServiceUri is { } nonNullOnionServiceUri && torManager.TorControlClient is {} nonNullTorControlClient)
+					{
+						await nonNullTorControlClient.DestroyHiddenService(nonNullOnionServiceUri.Host, CancellationToken.None).ConfigureAwait(false);
```

Comment:
Are there some scenarios where this can take some time and delay software closure? If the tor instance is unresponsive etc...

Reply:
Umm it could be but I have tested it and it does its job extremely fast.